### PR TITLE
Add Application content

### PR DESCRIPTION
**Pull request by dturanski**
Addresses https://github.com/spring-cloud/spring-cloud-dataflow/issues/3883 
Still need to remove content from reference.

I think we should move the Application Metadata section under Applications @sabbyanandan @tzolov

### CODE REVIEW

File: content/documentation/pages/9-applications/0-index.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
+---
+path: 'applications/'
+title: 'Applications'
+description: 'Using stream and task applications'
+summary: true
+---
+
+# Using Applications with Spring Cloud Data Flow
+
+Spring Cloud Data Flow provides native support for applications built with and [Spring Cloud Stream](https://spring.io/projects/spring-cloud-stream) or [Spring Cloud Task](https://spring.io/projects/spring-cloud-stream).
```
Comment (ilayaperumalg) [inline content/documentation/pages/9-applications/0-index.md:10]:
```suggestion
Spring Cloud Data Flow provides native support for applications built with [Spring Cloud Stream](https://spring.io/projects/spring-cloud-stream) or [Spring Cloud Task](https://spring.io/projects/spring-cloud-stream).
```

File: content/documentation/pages/9-applications/0-index.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+Note, however, that, once downloaded, applications may be cached locally on the Data Flow server, based on the resource location. If the resource location does not change (even though the actual resource bytes may be different), then it is not re-downloaded. When using maven:// resources on the other hand, using a constant location may still circumvent caching (if using -SNAPSHOT versions).
+
+Moreover, if a stream is already deployed and using some version of a registered app, then (forcibly) re-registering a different app has no effect until the stream is deployed again.
+
+<!--END_CAUTION-->
+
+<!--TIP-->
+
+In some cases, the Resource is resolved on the server side. In others, the URI is passed to a runtime container instance where it is resolved. Consult the specific documentation of each Data Flow Server for more detail.
```
Comment (ilayaperumalg) [inline content/documentation/pages/9-applications/0-index.md:83]: Can we be more specific about this? Also, `documentation of each Data Flow Server` is confusing.

File: content/documentation/pages/9-applications/1-pre-packaged-applications.md [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+- Are available as Apache Maven artifacts or Docker images
+- Use RabbitMQ or Apache Kafka
+- Support monitoring through [Prometheus](https://prometheus.io/) and [InfluxDB](https://www.influxdata.com/)
+- Contain metadata for application properties used in the UI and code completion in the shell
+
+You can register stream and task applications by using the Data Flow UI or the shell.
+
+You can register applications individually by using the `app register` command or in bulk by using the `app import` command.
+
+For streams, depending on whether you use Kafka or RabbitMQ, you can register the applications by using the respective URLs:
```
Comment (ilayaperumalg) [inline content/documentation/pages/9-applications/1-pre-packaged-applications.md:29]: I think we can show the `docker` based ones as well along with the `maven`.

File: content/documentation/pages/9-applications/1-pre-packaged-applications.md [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+| [triggertask](%stream-applications-doc%/#spring-cloud-stream-modules-triggertask-source)       |                                                                                                                             |                                                                                                              |
+| [twitterstream](%stream-applications-doc%/#spring-cloud-stream-modules-twitterstream-source)   |                                                                                                                             |                                                                                                              |
+
+## Task Applications
+
+The Spring team develops and maintains [task/batch applications](https://cloud.spring.io/spring-cloud-task-app-starters/) and publishes these applications to the [Spring public Maven repository](https://repo.spring.io/release/org/springframework/cloud/task/app/) and to [Dockerhub](https://hub.docker.com/u/springcloudtask) in accordance with a planned release schedule, normally following significant Spring Boot, Spring Cloud Task, or Spring Batch releases.
```
Comment (ilayaperumalg) [inline content/documentation/pages/9-applications/1-pre-packaged-applications.md:89]: We can say that the out of the box `task` apps are for `demo` purposes as one has to write the batch/task apps based on the use cases.

File: content/documentation/pages/9-applications/1-pre-packaged-applications.md [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+Spring Cloud Data Flow supports bulk registration of applications, via a standard properties file format.
+For convenience, we publish static properties files with application-URIs (for either maven or docker) for all the out-of-the-box stream and task/batch apps.
+You can use these files in Spring Cloud Data Flow to register all the application-URIs in bulk.
+Alternately, you can register them individually or provide your own custom property file with only the required application-URIs in it.
+Bulk registration is convenient for getting started with SCDF.
+To reduce clutter, it is recommended, to maintain a “focused” list of desired application-URIs in a custom property file.
+
+# Supported Spring Cloud Stream Applications
+
+The following table includes the dataflow.spring.io links to the Supported _Stream_ Applications
```
Comment (ilayaperumalg) [inline content/documentation/pages/9-applications/1-pre-packaged-applications.md:107]: Missing hyperlink?

File: yarn.lock [hunk 6]
```diff
@@ -25,7 +25,16 @@
     invariant "^2.2.4"
     semver "^5.5.0"
 
-"@babel/core@^7.2.2", "@babel/core@^7.8.4", "@babel/core@^7.8.7":
+"@babel/compat-data@^7.9.6":
```
Comment (ilayaperumalg) [inline yarn.lock:28]: We can remove this file.

**Comment by ilayaperumalg** [comment]
@dturanski The content looks great!

File: content/documentation/pages/9-applications/1-pre-packaged-applications.md (hunk 4 above)
Comment (dturanski) [reply content/documentation/pages/9-applications/1-pre-packaged-applications.md:89]: I was thinking about that, but will not be entirely true when Michael adds the new batch app.

File: content/documentation/pages/9-applications/0-index.md (hunk 2 above)
Comment (dturanski) [reply content/documentation/pages/9-applications/0-index.md:83]: I'm removing this. I'm not sure what it means and is not really helpful.

**Comment by ilayaperumalg** [comment]
LGTM

**Pull request merged**